### PR TITLE
feat: add custom search to InternalDropdown and Picklist

### PR DESCRIPTION
**Pull request by HellWolf93**
## Changes proposed in this PR:
- add onSearch prop to InternalDropdown and Picklist

- [X] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

### CODE REVIEW

File: src/components/InternalDropdown/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    /** If is set to true, then a search input to filter is showed. */
+    enableSearch: boolean;
+    /** Specifies that multiple items can be selected */
+    multiple: boolean;
+    /** Show checkbox */
+    showCheckbox: boolean;
+    /** The text to show on the header when showCheckbox is true */
+    placeholder: string;
+    /** Action triggered when search query changes */
+    onSearch: (query: string) => void;
```
Comment (LeandroTorresSicilia) [inline src/components/InternalDropdown/index.d.ts:34]: pass `?` to all props

**Review by TahimiLeonBravo — APPROVED** [review]
+1

File: src/components/InternalDropdown/index.d.ts [hunk 2]
```diff
@@ -32,4 +32,6 @@ export interface InternalDropdownProps extends BaseProps {
     placeholder?: string;
     /** Action triggered when search query changes */
     onSearch?: (query: string) => void;
+    /** When true, the onSearch callback will be debounced */
+    debounce: boolean;
```
Comment (LeandroTorresSicilia) [inline src/components/InternalDropdown/index.d.ts:36]: let's put it as optional

File: src/components/Picklist/index.d.ts [hunk 3]
```diff
@@ -30,6 +30,7 @@ export interface PicklistProps extends BaseProps {
     disabled?: boolean;
     readOnly?: boolean;
     id?: string;
+    debounce: boolean;
```
Comment (LeandroTorresSicilia) [inline src/components/Picklist/index.d.ts:33]: same ^^

**Review by LeandroTorresSicilia — APPROVED** [review]

**Review by LeandroTorresSicilia — APPROVED** [review]

**Pull request merged**